napari/_qt/qt_main_window.py

```diff
@@ -66,12 +67,26 @@ def __init__(self, parent=None) -> None:
         self._preferences_dialog_size = QSize()
         self._status_bar = self.statusBar()
 
-        # set SETTINGS plugin defaults.
+        print('call order', SETTINGS.plugins.call_order)
+        print('state', SETTINGS.plugins.disabled_plugins)
+
+        # set the call order default before anything else is set.
         plugins.load_settings_plugin_defaults(SETTINGS)
 
-        # set the values in plugins to match the ones saved in SETTINGS
+        # if there are any disabled plugins on start up saved in settings,
+        # need to disable them in the plugin manager.
+        if len(SETTINGS.plugins.disabled_plugins) > 0:
+
+            for name in SETTINGS.plugins.disabled_plugins:
+                plugins.plugin_manager.set_blocked(name, True)
```

Comment:
that's right, you don't need both.  I should have cleaned this up.  I guess thinking about it though, I would prefer to keep this one and not the one in the `plugins.__init__ `, because I had to import SETTINGS in the init file just for that one line.  I would prefer not to do that, however, if you believe it belongs in the init file, I will keep it that way.  I'm not sure how much that matters, but I was thinking it was better not to import SETTINGS more places than needed.  What do you think @tlambert03 ?

Reply:
I think I'm personally a bit less concerned with imports (short of circular import problems of course) than I am about spreading the business logic for plugins across multiple modules.
Since it's really important that we update the `plugin_manager._blocked` list _before_ discovery happens (lest all of those plugins will be imported), I definitely prefer this update to happen over in `plugins.__init__`.   

Also, though we definitely don't use it often, it's important to kinda keep the "headless" use case of napari in the back of our minds.  That is: there should be no assumption that a `QMainWindow` is even going to be created when using napari (in fact, the majority of our test suite [can pass even if Qt is not installed](https://github.com/napari/napari/runs/2482817391) in the environment).  So we also shouldn't rely on the creation of a Qt window for something as fundamental as blocking package discovery.  (same goes for call_order really... imagine a script that wants to use napari reader plugins, but doesn't really care whether a main window is created or not)